src/components/Users.tsx

```diff
@@ -1,12 +1,137 @@
-import React from 'react';
-import { Container } from 'react-bootstrap';
+import React, { useState, useContext, useEffect } from 'react';
+import { Col, Container, Row, Table, Button } from 'react-bootstrap';
 import styled from 'styled-components';
-import img from '../img/welcome.jpg';
+import { v4 as uuidv4 } from 'uuid';
+import { useNotify } from '../hooks/notify.hook';
+import AddFormModal from './modals/AddFormModal';
+import DeleteFormModal from './modals/DeleteFormModal';
+import { ISaveUserResponse } from '../shared/interfaces/common';
+import { Notify } from './Notify';
+import { useUsers } from '../hooks/users.hook';
+import { AuthContext } from '../context/AuthContext';
+import {
+    saveUserController,
+    getUsersController,
+} from '../controllers/users.controller';
+
+const TableWrapper: any = styled.div`
+    background: #fff;
+    border-radius: 3px;
+    box-shadow: 0 1px 1px rgba(0, 0, 0, 0.05);
+`;
+const TableTitle: any = styled.div`
+    padding-bottom: 15px;
+    background: #435d7d;
+    color: #fff;
+    padding: 16px 30px;
+    border-radius: 3px 3px 0 0;
+`;
+const TableTitleH2: any = styled.h2`
+    margin: 5px 0 0;
+    font-size: 24px;
+`;
+const ModalWrapper: any = styled.div`
+    & button:first-child {
+        margin-right: 20px;
+    }
+`;
+const ActionsWrapper: any = styled.div`
+    display: flex;
+    width: 50%;
+    justify-content: space-around;
+`;
 
 const Users = () => {
+    const { message, type, addNotification, show, setShow } = useNotify();
+    const { users, addUsers } = useContext(AuthContext);
+
+    const TableHeader = () => {
+        const HEADERS: string[] = ['User ID', 'User Name', 'Age', 'Actions'];
+        const headerList = HEADERS.map((header) => (
+            <th key={uuidv4()}>{header}</th>
```

Comment:
For keys for similar JSX elements you can use item indexes from array. Sometimes you can add some specific string prefix. One more thing: no need to create unique index across all application, you can create unique index in the scope of current component only